teacherchannel/module.py

```diff
@@ -0,0 +1,248 @@
+import nextcord
+from nextcord.ext import tasks, commands
+
+import pie.database.config
+from pie import check, i18n, logger, utils
+
+from .database import TeacherChannel as TeacherChannelDB
+
+_ = i18n.Translator("modules/school").translate
+guild_log = logger.Guild.logger()
+config = pie.database.config.Config.get()
+
+
+class TeacherChannel(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.load_deltas.start()
+
+    @commands.guild_only()
+    @commands.check(check.acl)
+    @commands.group(name="teacherchannel")
+    async def teacherchannel_(self, ctx):
+        """Syncing teacher channels' permissions."""
+        await utils.discord.send_help(ctx)
+
+    @commands.check(check.acl)
+    @teacherchannel_.command(name="list")
+    async def teacherchannel_list(self, ctx):
+        """List active teacher channels."""
+
+        class Item:
+            def __init__(self, obj: TeacherChannelDB):
+                channel_o = ctx.guild.get_channel(obj.master_id)
+                channel_t = ctx.guild.get_channel(obj.slave_id)
+                self.o_channel = "#" + getattr(channel_o, "name", str(obj.master_id))
+                self.t_channel = "#" + getattr(channel_t, "name", str(obj.slave_id))
+                self.teachers = ", ".join(
+                    [
+                        getattr(
+                            ctx.guild.get_member(i.user_id),
+                            "display_name",
+                            str(i.user_id),
+                        )
+                        for i in obj.teachers
+                    ]
+                )
+
+        items = [Item(i) for i in TeacherChannelDB.get_all(ctx.guild.id)]
+        if len(items) < 1:
+            await ctx.reply(_(ctx, "No teacher channels are set."))
+            return
+        table = utils.text.create_table(
+            items,
+            header={
+                "o_channel": _(ctx, "Original channel"),
+                "t_channel": _(ctx, "Teacher channel"),
```

Comment:
Ah, it's `original` and `teacher`. Maybe `master` and `slave`? Or `main` and `dependent`, if we really want to follow the conscious language movement.